src/server.rs

```diff
@@ -74,10 +77,10 @@ impl Server {
     }
 
     /// Decrypt and deserialize
-    fn deserialize_share(&self, encrypted_share: &[u8]) -> Result<Vec<Field>, ServerError> {
+    fn deserialize_share(&self, encrypted_share: &[u8]) -> Result<Vec<F>, ServerError> {
         let share = decrypt_share(encrypted_share, &self.private_key)?;
         Ok(if self.is_first_server {
-            deserialize(&share)
+            deserialize(&share).or_else(|err| Err(ServerError::Serialize(err)))?
```

Comment:
```suggestion
            deserialize(&share)?
```
[`?` will automatically use an appropriate `From` implementation to convert errors to the appropriate type](https://doc.rust-lang.org/stable/reference/expressions/operator-expr.html#the-question-mark-operator) and `thiserror` made you a `From` due to `Serialize(#[from] SerializeError)` so you don't need to wrap the error yourself.

Reply:
Also magic. Also done.